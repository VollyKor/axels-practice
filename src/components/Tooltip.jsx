import { Overlay, Popover } from 'react-bootstrap';

export default function Tooltip({
    fieldName = '',
    forwardRef,
    touched,
    errors,
}) {
    return fieldName && forwardRef && touched && errors ? (
        <Overlay
            target={forwardRef?.current}
            show={touched[fieldName] && !!errors[fieldName]}
            placement="top"
        >
            <Popover className="bg-white text-danger p-1 px-3 border-none shadow">
                {errors.fullName}
            </Popover>
        </Overlay>
    ) : (
        'Error, Some props are missing fro Tooltip'
    );
}
