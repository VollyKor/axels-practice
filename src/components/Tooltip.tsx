import { Overlay, Popover } from 'react-bootstrap';

interface Props {
    fieldName: FieldNAmeTypes | '';
    forwardRef: React.MutableRefObject<any>;
    touched: TformikTouched;
    errors: TformikErrors;
}

const Tooltip = ({ fieldName = '', forwardRef, touched, errors }: Props) => {
    return fieldName && forwardRef && touched && errors ? (
        <Overlay
            target={forwardRef?.current}
            show={touched[fieldName] && !!errors[fieldName]}
            placement="top"
        >
            <Popover
                id={`${fieldName} ${Date.now()}`}
                className="bg-white text-danger p-1 px-3 border-none shadow"
            >
                {errors[fieldName]}
            </Popover>
        </Overlay>
    ) : (
        <div>'Error, Some props are missing from Tooltip'</div>
    );
};

export default Tooltip;
