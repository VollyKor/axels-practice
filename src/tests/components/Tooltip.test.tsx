import { shallow } from 'enzyme';

import Tooltip from 'components/Tooltip';

it('renders snapshot', () => {
    //@ts-ignore
    const wrapper = shallow(<Tooltip />);
    expect(wrapper).toMatchSnapshot();
});
