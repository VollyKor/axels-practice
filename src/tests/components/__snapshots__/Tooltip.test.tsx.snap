// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders snapshot 1`] = `
<div>
  'Error, Some props are missing from Tooltip'
</div>
`;
